source/slang/slang-parser.cpp

```diff
@@ -557,6 +557,16 @@ namespace Slang
         return false;
     }
 
+    bool AdvanceIf(Parser* parser, TokenType tokenType, Token* outToken)
+    {
+        if (parser->LookAheadToken(tokenType))
+        {
+            *outToken = parser->ReadToken();
+            return true;
+        }
+        return false;
+    }
+
```

Comment:
Not really related to this change, but: it would be nice in theory if `AdvanceIf` *always* returned a `Token`, and then we could have a custom `bool` conversion where an invalid token evaluates as `false` and any valid token evaluates as `true`.

The same could be done for the functions like `ReadToken()`  so that when the expected token isn't there, it can return a "falsey" invalid token, allowing parser code to detect and handle certain cases of errorneous input more succintly.

Making `Token` convertible to `bool` is probably not a step to take lightly, so I wouldn't propose adding it to another change like this. (Realistically, we'd probably want a custom wrapper *subtype* of `Token` that has the conversion, so that we don't accidentally make all `Token`s convertible to `bool` by accident).